### DISCUSSION TITLE
I made a chrome extension that creates/copies a fabric command for you

### DISCUSSION THREAD

**Discussion by stuboo** [danielmiessler/Fabric#756] [Show and tell]
I found that I was often using the `extract_wisdom` pattern and it occurred to me that I could improve my workflow by creating a simple Chrome extension. 

This extension is designed to simplify the process of capturing YouTube video URLs and preparing them for use with Fabric. I use it to create obsidian notes using the `extract_wisdom` pattern. Just install the extension, navigate to a YouTube video, and click the icon. The extension will ask you for a filename and confirm when the fabric command has been copied to your clipboard.

Example:
`
yt https://www.youtube.com/watch?v=ABK00e2XdPo --transcript | fabric --pattern extract_wisdom | save multimodal_rag_ABK00e2XdPo -s -n
`

### Features:

- **Capture YouTube URLs**: Automatically captures the URL of the YouTube video you're watching.
- **Filename Input**: Provides a user-friendly modal to enter a custom filename.
- **Formatted Command**: Copies a formatted command to the clipboard, ready for use with Fabric.
- **Feedback Message**: Displays the copied text for confirmation.
- **Close Button**: Easily close the modal after copying the URL.

### How It Works:

1. **Visit a YouTube video URL**: For example, `https://www.youtube.com/watch?v=ABK00e2XdPo&t=21s`.
2. **Click the ytFabric extension button**: A modal will appear requesting a filename.
3. **Enter a filename**: For instance, enter `my_video`.
4. **Submit the filename**: Click the "Submit" button.
5. **Clipboard content**: The clipboard will now contain the following text:

    `yt https://www.youtube.com/watch?v=ABK00e2XdPo --transcript | fabric --pattern extract_wisdom | save my_video_ABK00e2XdPo -s -n`

6. **Feedback**: A message showing the copied text will be displayed in the modal.
7. **Close the modal**: Click the "Close" button to manually close the modal.

### Installation:

1. Clone the repository or download the ZIP file from [GitHub](https://github.com/stuboo/ytFabric) and extract it.
2. Open Chrome and navigate to `chrome://extensions/`.
3. Enable "Developer mode" in the top right corner.
4. Click "Load unpacked" and select the directory where you have the extension files.

### Icon Credit:

<a href="https://www.flaticon.com/free-icons/fabric" title="fabric icons">Fabric icons created by Smashicons - Flaticon</a>

I hope you find this extension as useful as I do!

**Comment by calebnewman** [comment]
This is great, thank you!

**Comment by fabioscarsi** [comment]
@stuboo, great job. Just one suggestion: the recently released official Obsidian Web Clipper (https://github.com/obsidianmd/obsidian-clipper) has a feature (still not released (https://github.com/obsidianmd/obsidian-clipper/issues/86#issuecomment-2421960446) to integrate LLMs in the flow. Considering that Obsidian Web Clipper is open source and that you have done excellent work with your Chrome extension, can you take a look and evaluate some convergence of the two projects or some contribution to the official Obsidian Web Clipper? (I'm just a user with no affiliation, but integrating Fabric patterns in the official Obsidian Web Clipper for any clipping would be an incredibly useful achievement ...). Thanks anyway for your extension.
